### PR TITLE
Update Octoliner API

### LINKED ISSUE #3: Update interface to match the Arduino library version
The Arduino-version of Octoliner library is about to be updated (for good reasons) and OctolinerPi should follow the change.

The minimal and enough interface is documented at https://github.com/amperka/Octoliner/blob/V2/API.md

Pi-version should have exactly the same methods and signatures (spelled in PEP8 though). This means adding a few methods, renaming few others, and removing the method about the brightness.

### Acceptance criteria

- [ ] The repository has `API.md` documenting the Python API
- [ ] The `Octoliner` class methods match those found in Arduino Octoliner v2

### PR DESCRIPTION
**Pull request by AlexErAmp**
Close #3

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+Sets the sensitivity of the photodetectors in the range from `0` to `1.0`.
+
+### `analog_read(sensor: int) -> float`
+
+Reads the value from one line sensor. Return value in range from `0` to `1.0`.
+
+### `analog_read_all(analog_values: list) -> None`
+
+Reads all 8 channels to the `analog_values` list.
```
Comment (nkrkv) [inline API.md]: Not pythonic. Should be:

`analog_read_all() -> list(float)`

**Review by nkrkv — CHANGES_REQUESTED** [review]

File: API.md (hunk 1 above)
Comment (AlexErAmp) [reply API.md]: Should a function return a new list, and not modify an existing one?

Comment (AlexErAmp) [reply API.md]: Done.

**Review by nkrkv — APPROVED** [review]
:+1:

**Pull request merged**